### ISSUE TITLE
3.2.1: build depends on `setuptools` and `poetry`. Is that correct?🤔

### ISSUE THREAD

**Issue by kloczek** [graphql-python/graphql-core#178]
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting dependencies for wheel...

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pep517/wrappers.py", line 332, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pep517/in_process/_in_process.py", line 89, in _build_backend
    obj = import_module(mod_path)
  File "/usr/lib64/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'poetry'
```
However after adding `poetry-core` to build env  ..
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting dependencies for wheel...

ERROR Missing dependencies:
        setuptools>=59,<70
```
hmm 🤔

**Comment by kloczek** [comment]
Just tested that commit and with it still pep157 build requires both `setuptools` and `poetry-core`
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting dependencies for wheel...

ERROR Missing dependencies:
        setuptools>=59,<70
```

**Comment by Cito** [comment]
@kloczek Yes, #180 does not solve this issue.

GraphQL-py has both a setup.py and a pyproject.toml to support both ways of installation.

Will clean this up eventually and do whatever is currently best practice. If you have any links or recommendations let me know.

**Comment by Cito** [comment]
This is now fixed in the main branch (used for the upcoming 3.3 release), we only require poetry as build tool now.

**Comment by Cito** [comment]
Fix available in v3.3.0a1 for testing.
